GoogleForCreators/web-stories-wp
# Dashboard: Toaster UI

## Summary
Adds a Toaster UI component w/ context to generate and dismiss toasts

## Relevant Technical Choices
- New Component called `Toaster` controls message queue for active toasts 
- Context controls adding, removing and displaying
- Sets auto dismiss at 10000ms (10s) 
- Tracks all existing messages sent to toaster to prevent duplication of errors that are still present in redux or higher state. Toasts will reset on useEffect clean up with page refresh. 
- Moved the alert wrapper to toaster since the toaster is really the alert wrapper 🤷‍♀️  so I made some adjustments to the alert component/api - mostly just simplifying it. 

## To-do
- Wire up view 
- More tests (ran into a time suck issue with testing the interval timeout auto dismiss, commented that test out) 

## User-facing changes
None

## Testing Instructions
- See my unit tests 
- Use storybook. I have a demo set up in Dashboard/Components/Toaster that allows you to add 5 toasts. You can toggle the ability to early dismiss. These should timeout on their own in 10 seconds but you can dismiss them manually in any order you want to. 
- I also made the dismiss button optional so see that update reflected in Dashboard/Components/Alert

**Adding Toasts in Dashboard/Components/Toaster**
![trigger toasts](https://user-images.githubusercontent.com/10720454/85163614-f1445280-b217-11ea-9361-7ce8431e062d.gif)

**Auto Dismiss Toasts every 10s (full 10s not captured!)**
![auto dismiss toast](https://user-images.githubusercontent.com/10720454/85163599-ed183500-b217-11ea-91ae-ad7f89b12b85.gif)

**Manually Dismiss Toasts in custom order**
![manually remove toasts](https://user-images.githubusercontent.com/10720454/85163618-f30e1600-b217-11ea-95d2-536a3092dfee.gif)

---



Addresses #2262
